subprojects/core/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DependencyFactory.java

```diff
@@ -34,7 +35,8 @@
         }
     }
 
-    Dependency createDependency(Object dependencyNotation);
+    Dependency createDependency(Object dependencyNotation); //we should consider to change the return type to DirectDependency, which requires adjustment in Kotlin DSL
```

Comment:
@bamboo similar as the other comment: Would it be save to change the type here and rebuilt Koltin DSL? If I just change the type, I have failing Kotlin DSL tests in `integTest`.